apps/lib/publisher_agent/registry.go

```diff
@@ -30,14 +33,31 @@ func (c *RegistryClient) GetBrokersForPublisher(publisherKey signer.PublisherKey
 	if err != nil {
 		return nil, fmt.Errorf("failed to get broker list: %v", err)
 	}
+
 	var brokers []BrokerConnectionConfig
 	err = json.Unmarshal(response, &brokers)
 	if err != nil {
-		return nil, fmt.Errorf("failed to unmarshal broker list: %v", err)
+		// check if the registry returned an error
+		var errorResponse RegistryErrorResponse
```

Comment:
the rest query might succeed, but return an error message. We should make clear the error is clear (since it currently says "failed to unmarshal broker list" when the stork auth is incorrect)